internal/plugin/server.go

```diff
@@ -500,6 +503,10 @@ func (plugin *NvidiaDevicePlugin) apiDevices() []*pluginapi.Device {
 // updateResponseForDeviceListEnvvar sets the environment variable for the requested devices.
 func (plugin *NvidiaDevicePlugin) updateResponseForDeviceListEnvvar(response *pluginapi.ContainerAllocateResponse, deviceIDs ...string) {
 	response.Envs[plugin.deviceListEnvvar] = strings.Join(deviceIDs, ",")
+
+	if ids := plugin.rm.GetImexChannelIDs(); len(ids) > 0 {
```

Comment:
@klueska what is the expected behaviour if we are using `volume-mounts`?

Reply:
This is specific to the "legacy", i.e. non-CDI case.

I guess it makes sense to have a separate container path mounted from /dev/null on the host for the IMEX channels, similar to the GPUs. SOmething like `/var/run/nvidia-imex-channels/<channel-num>` and then set the envvar to this path like we do here: https://github.com/NVIDIA/k8s-device-plugin/blob/966d5b8fe63bfb09ac26b46404e990bf3b5808e1/internal/plugin/server.go#L507

Reply:
I suppose that in this case we would probably want to ALSO set the `NVIDIA_IMEX_CHANNELS` envvar.

Reply:
That's what I mean by "and then set the envvar to this path like we do here..."

Reply:
> I guess it makes sense to have a separate container path mounted from /dev/null on the host for the IMEX channels, similar to the GPUs. Something like /var/run/nvidia-imex-channels/<channel-num> and then set the envvar to this path like we do here:

OK. This would requires explicit support in the NVIDIA Container Runtime. As long as we're ok introducing that requirement. 

My suggestion was that we do **the same thing** as what we do for the `envvar` case. That is to say, when `volume-mounts` is selected, we would request devices via `/dev/null` mounts and use the standard legacy mode for IMEX channels. This is probably not the right way to go though.